File: android/src/main/java/org/maplibre/reactnative/components/layers/types/heatmap/MLRNHeatmapLayerManager.java
package org.maplibre.reactnative.components.layers.types.heatmap;

import com.facebook.react.uimanager.ThemedReactContext;

import org.maplibre.reactnative.components.layers.MLRNVectorLayerManager;

public class MLRNHeatmapLayerManager extends MLRNVectorLayerManager<MLRNHeatmapLayer> {
    public static final String REACT_CLASS = "MLRNHeatmapLayer";

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    protected MLRNHeatmapLayer createLayerViewInstance(ThemedReactContext reactContext) {
        return new MLRNHeatmapLayer(reactContext);
    }
}

File: android/src/main/java/org/maplibre/reactnative/components/layers/types/heatmap/MLRNHeatmapLayerManager.kt
package org.maplibre.reactnative.components.layers.types.heatmap

import com.facebook.react.uimanager.ThemedReactContext
import org.maplibre.reactnative.components.layers.MLRNVectorLayerManager

class MLRNHeatmapLayerManager : MLRNVectorLayerManager<MLRNHeatmapLayer?>() {
    override fun getName(): String {
        return REACT_CLASS
    }

    override fun createLayerViewInstance(reactContext: ThemedReactContext?): MLRNHeatmapLayer {
        return MLRNHeatmapLayer(reactContext)
    }

    companion object {
        const val REACT_CLASS: String = "MLRNHeatmapLayer"
    }
}
